_posts/2016-07-25-Web-of-trusted-commits.md

```diff
@@ -0,0 +1,138 @@
+
+The following command allows any individual with bad intentions to commit (malicious) code under your name, meaning that you will get the blame for the backdoor or exploit "you" committed:
+
+  {% highlight bash %}
+  # Individual commit.
+  $ git commit -a -m "a message" --author "Sherlock H. <sherlock.h@bakerstreet.org>"
+
+  # Global settings.
+  $ git config --global user.name 'Sherlock H.'
+  $ git config --global user.email sherlock.h@bakerstreet.org
+  {% endhighlight %}
+
+## Ensuring Trust
+
+This blog post tells the story of Sherlock H. Sherlock is a witty developer who holds any security-related topic very close to his heart. After a fair amount of pondering about how he could solve the problem of black-hearted developers impersonating his personality, he decided to add **a Digital Signature** to his commits. By adding a signature Sherlock can finally sleep soundly at night because the signature indicates that he really issued the commit and that it has not been tampered with since he sent it. Moreover it can be used to trace the origin of malicious code that has made its way into a repository. The signature makes it also difficult for the signer to deny having signed something because the Digital Signature is unique to both the commit and the signer, and binds them together. Sherlock can now wholeheartedly vouch for the commit.
+
+Consider the following scenario:
+
+> Sherlock wants to send an urgent message to his fellow developer John W. telling that their application has been compromised by Jim M, a criminal mastermind who only has unkind intentions. John wants the guarantee that the message he received is sent by Sherlock and has not been tampered with by Jim.
+
+In order to securely exchange messages, both Sherlock and John will make use of their **Key Pairs**. A Key Pair consists of a **Public and Private Key** which are two unique mathematically related cryptographic keys. As its name suggests, the Public Key is made available to everyone by handing out copies or sharing them through a publicly accessible repository. The Private Key however must be kept confidential to its respective owner.
+
+Sherlock and John can do the following with the use of their Key Pair:
+
++ **Signing**
+  - The message is still readable to everyone.
+  - Guarantee of the sender's identity (aka Sherlock).
+  - Guarantee that the message has not been tampered with since it has been signed by the sender (aka Sherlock).
+
++ **Encryption**
+  - The message is only readable by the designated recipient (aka John).
+  - No guarantee of the sender's identity (aka Sherlock).
+  - Encryption can be done symmetrically by using a Shared Secret Key, a single key is then used for both encryption and decryption. Asymmetrical encryption (aka Public Key encryption) with a Public/Private Keypair uses one key for encryption and another for decryption. Note that the advantages and challenges of using either encryption type is beyond the scope of this blog post.
+

```

Comment:
You can also give examples of both encryption types. I think that the Belgian eID uses asymmetrical encryption.

Maybe also highlight symmetrical en asymmetrical to stress that there are two types?